src/include/optimizer/statistics/column_stats.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <sstream>
+
+#include "catalog/catalog_defs.h"
+#include "common/macros.h"
+
+namespace terrier::optimizer {
+
+class ColumnStats {
+ public:
+  ColumnStats(catalog::db_oid_t database_id, catalog::table_oid_t table_id, catalog::col_oid_t column_id,
+              const std::string &column_name, bool has_index, size_t num_rows,
```

Comment:
Also, I don't think we should store `has_index` here. How is it being used?

This flag is vague and not helpful since the index can be dropped and this object won't know.